src/D2L.Security.OAuth2/Keys/Development/InMemoryPublicKeyDataProvider.cs

```diff
@@ -14,30 +15,34 @@ namespace D2L.Security.OAuth2.Keys.Development {
 	public sealed class InMemoryPublicKeyDataProvider : IPublicKeyDataProvider {
 		private readonly ConcurrentDictionary<Guid, JsonWebKey> m_keys = new ConcurrentDictionary<Guid, JsonWebKey>();
 
+		[GenerateSync]
 		Task<JsonWebKey> IPublicKeyDataProvider.GetByIdAsync( Guid id ) {
 			if( !m_keys.TryGetValue( id, out JsonWebKey key ) ) {
 				return Task.FromResult<JsonWebKey>( null );
 			}
 			return Task.FromResult( key );
 		}
 
+		[GenerateSync]
 		Task<IEnumerable<JsonWebKey>> IPublicKeyDataProvider.GetAllAsync() {
 			IEnumerable<JsonWebKey> result =
 				new ReadOnlyCollection<JsonWebKey>( m_keys.Values.ToList() );
 
 			return Task.FromResult( result );
 		}
 
-		Task IPublicKeyDataProvider.SaveAsync( Guid id, JsonWebKey key ) {
+		[GenerateSync]
+		async Task IPublicKeyDataProvider.SaveAsync( Guid id, JsonWebKey key ) {
 			if( !m_keys.TryAdd( id, key ) ) {
 				throw new InvalidOperationException( "Attempted to add a key twice" );
 			}
-			return Task.CompletedTask;
+			await Task.CompletedTask.ConfigureAwait(false);
 		}
 
-		Task IPublicKeyDataProvider.DeleteAsync( Guid id ) {
+		[GenerateSync]
+		async Task IPublicKeyDataProvider.DeleteAsync( Guid id ) {
 			m_keys.TryRemove( id, out JsonWebKey removedKey );
-			return Task.CompletedTask;
+			await Task.CompletedTask.ConfigureAwait(false);

```

Comment:
Hm, we could transform `return Task.CompletedTask;` into just `return;` I think.

[Check out the decompilation of](https://sharplab.io/#v2:EYLgtghglgdgPgAQAwAIEEYCsBuAsAKAIQGY10A2NAJhQGEUBvAlFtUjShADjUoDEA9gIAUASkbNWUhAE5eAOloCwABwA2AUwAuGgCYJyigTABmUAOYBXAE4aAggHdoW4ShMQ1AZw0pRefFIAvgSBQA=) a method with `await Task.CompletedTask.ConfigureAwait( false );`. It's ultimately not really doing much else but it's definitely more junk than just `return Task.CompletedTask`. Mostly though it's just "weird" to see `await Task.CompletedTask` :) I'd like for `[GenerateSync]` to allow us to write idiomatic code.

Reply:
I wouldn't add the `async` modifier to them, e.g. I'd hope that

```cs
Task IPublicKeyDataProvider.DeleteAsync( Guid id ) {
	m_keys.TryRemove( id, out JsonWebKey removedKey );
	return Task.CompletedTask;
}
```

could get transformed into

```cs
void IPublicKeyDataProvider.Delete( Guid id ) {
	m_keys.TryRemove( id, out JsonWebKey removedKey );
	return;
}
```

or something like that